pkg/tokenutil/internal/algorithm/algorithm.go

```diff
@@ -0,0 +1,46 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package algorithm
+
+import (
+	"github.com/pkg/errors"
+)
+
+// SignatureAlgorithm is a collection of all signature algorithm and it provides
+// some basic method to use.
+type SignatureAlgorithm interface {
+	// IsHMAC is used to checking if the algorithm is belong to HMAC.
+	IsHMAC() bool
```

Comment:
why do we need this method? All the algorithms we supported return `true`.

Reply:
Maybe I should add this method in another pull request. When I create a private or public key, I will add other signature algorithms so I need to add this to distinguish the signature algorithm.

Reply:
Please do it in a separate pull request.